apps/root/src/services/priceService.ts

```diff
@@ -183,4 +191,59 @@ export default class PriceService {
 
     return graphData;
   }
+
+  async getPricesForTokenGraph({ token, span, period }: { token: Token; span: number; period: TimeString }) {
+    const wrappedProtocolToken = getWrappedProtocolToken(token.chainId);
+
+    const adjustedToken =
+      token.address === PROTOCOL_TOKEN_ADDRESS
+        ? wrappedProtocolToken
+        : { ...token, address: getTokenAddressForPriceFetching(token.chainId, token.address) };
+
+    const tokens = [{ chainId: token.chainId, token: adjustedToken.address }];
+    try {
+      const prices = await timeoutPromise(
+        this.sdkService.sdk.priceService.getChart({
```

Comment:
this should be a method of the sdkService itself instead of accesing the sdk property